### PR TITLE
Further spot performance improvements

### PR DESCRIPTION
**Pull request by mpenet**
Small gains probably, I didn't measure yet. 
Even if some of the modifications do not show a lot of gains in pure speed it should be more "garbage friendly". 
This is just "surface" work, I am sure there are more gains to be had by changing how things are done in some places, but I am not so familiar with the codebase yet nor do I have a lot of free time at the moment :baby_bottle: :baby:

### CODE REVIEW

File: src/com/walmartlabs/lacinia/schema.clj [hunk 1]
```diff
@@ -445,12 +445,11 @@ z
           resolver-errors (when is-tuple?
                             (resolve-errors raw-value))
           enforced-tuple (enforcer [enforce-value])
-          all-errors (->> (concat
-                            (ensure-seq resolver-errors)
-                            (ensure-seq (second enforced-tuple)))
-                          (filter some?)
-                          (mapcat assert-and-wrap-error)
-                          seq)]
+          all-errors (->> (sequence (comp cat
```
Comment (mpenet) [inline src/com/walmartlabs/lacinia/schema.clj]: might be faster with `into`, depending if we care about laziness

**Comment by bcarrell** [comment]
Appreciate you taking a look at possibilities for improving performance -- I'll try and add some measurements.

**Comment by hlship** [comment]
I'm running a check on performance right now.  Getting rid of some reflection during parsing could show up.

**Comment by hlship** [comment]
Big drop in parse time.
parse-time
Small up-tick in execution time:

exec-time

**Comment by bcarrell** [comment]
Looks good to me then.  I'm fine with merging this.

**Comment by hlship** [comment]
I've merged this, along with updated benchmarks.

**Pull request closed without merging**